File: software/examples/csharp/ExampleCounterCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Hall Effect Bricklet 2.0

	// Callback function for counter callback
	static void CounterCB(BrickletHallEffectV2 sender, long counter)
	{
		Console.WriteLine("Counter: " + counter);
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletHallEffectV2 he = new BrickletHallEffectV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Configure counter with ±3000µT threshold and 10ms debounce
		he.SetCounterConfig(3000, -3000, 10000);

		// Register counter callback to function CounterCB
		he.CounterCallback += CounterCB;

		// Set period for counter callback to 0.1s (100ms)
		he.SetCounterCallbackConfiguration(100, true);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_counter_callback.rs
use std::thread;
use std::{error::Error, io};
use tinkerforge::{hall_effect_v2_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Hall Effect Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let he = HallEffectV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Configure counter with ±3000µT threshold and 10ms debounce
    he.set_counter_config(3000, -3000, 10000);

    let counter_receiver = he.get_counter_callback_receiver();

    // Spawn thread to handle received callback messages.
    // This thread ends when the `he` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for counter in counter_receiver {
            println!("Counter: {}", counter);
        }
    });

    // Set period for counter callback to 0.1s (100ms).
    he.set_counter_callback_configuration(100, true);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
